lib/ansible/modules/network/ios/ios_config.py

```diff
@@ -434,7 +449,15 @@ def main():
         contents = get_config(module, flags=flags)
         config = NetworkConfig(indent=1, contents=contents)
         if module.params['backup']:
-            result['__backup__'] = contents
+            if module.params['remove_extraneous']:
+                # remove configuration intro lines from running-config
+                cfg_build_obj = re.search(r'Building configuration.*(\n*)', contents)
+                cfg_current_obj = re.search(r'Current configuration.*(\n*)', contents)
+                if cfg_build_obj and cfg_current_obj:
+                    contents = contents.strip(cfg_build_obj.group() + cfg_current_obj.group())
```

Comment:
This seems overly complicated. You already have a regex, why not do `contents = re.sub('Building configuration.*\n*Current configuration.*\n*', '', contents)` instead of two searches and then a replace?

Reply:
makes sense. Thanks have updated the code.